jwst/assign_wcs/util.py

```diff
@@ -118,7 +118,8 @@ def wcs_from_footprints(dmodels, refmodel=None, transform=None, bounding_box=Non
         rotation = astmodels.AffineTransformation2D(np.array(wcsinfo['PC']))
         transform.append(rotation)
         if sky_axes:
-            scale = wcsinfo['CDELT'][sky_axes].mean()
+            cdelt1, cdelt2 = wcsinfo['CDELT']
```

Comment:
If you intend to use the CDELT corresponding to the celestial axes you should use

`cdelt1, cdelt2 = wcsinfo['CDELT'][sky_axes]`

This ensures that you are always ppicking the celestial axes regardless of the order of axes.